src/manager/component_utils/sc_component_utils.cpp

```diff
@@ -0,0 +1,88 @@
+#include <sc-memory/sc_addr.hpp>
+#include <sc-memory/sc_type.hpp>
+#include <sc-memory/sc_iterator.hpp>
+#include "sc_component_utils.hpp"
+#include "sc-agents-common/utils/CommonUtils.hpp"
+#include "sc-agents-common/utils/IteratorUtils.hpp"
+#include "../commands/keynodes/ScComponentManagerKeynodes.hpp"
+
+namespace utils
+{
+
+/**
+ * @brief Get content of component address link content
+ * @param context current sc-memory context
+ * @param componentAddr sc-addr of the component to install
+ * @return string of component address
+ */
+std::string ComponentUtils::GetComponentAddress(ScMemoryContext * context, ScAddr const & componentAddr)
+{
+  ScAddr componentAddressAddr;
+  std::string componentAddress;
+  ScIterator5Ptr componentAddressIterator = context->Iterator5(
+      componentAddr,
+      ScType::EdgeDCommonConst,
+      ScType::LinkConst,
+      ScType::EdgeAccessConstPosPerm,
+      keynodes::ScComponentManagerKeynodes::nrel_component_address);
+
+  if (componentAddressIterator->Next())
+  {
+    componentAddressAddr = componentAddressIterator->Get(2);
+    componentAddress = utils::CommonUtils::getLinkContent(context, componentAddressAddr);
+  }
+
+  return componentAddress;
+}
+
+/**
+ * @brief Get component dependencies vector
+ * @param context current sc-memory context
+ * @param componentAddr sc-addr of the component to install
+ * @return ScAddrVector consists of component dependencies sc-addrs,
+ * return empty vector if component has no dependencies
+ */
+ScAddrVector ComponentUtils::GetComponentDependencies(ScMemoryContext * context, ScAddr const & componentAddr)
+{
+  ScAddrVector componentDependencies;
+  ScIterator5Ptr const componentDependenciesIterator = context->Iterator5(
+      componentAddr,
+      ScType::EdgeDCommonConst,
+      ScType::NodeConst,
+      ScType::EdgeAccessConstPosPerm,
+      keynodes::ScComponentManagerKeynodes::nrel_component_dependencies);
+
+  while (componentDependenciesIterator->Next())
+  {
+    ScAddr componentDependenciesSet = componentDependenciesIterator->Get(2);
+    ScAddrVector componentCurrentDependencies =
+        utils::IteratorUtils::getAllWithType(context, componentDependenciesSet, ScType::NodeConst);
+    componentDependencies.insert(
+        componentDependencies.cend(), componentCurrentDependencies.cbegin(), componentCurrentDependencies.cend());
+  }
```

Comment:
Move declaration from cycle
```suggestion
  ScAddr componentDependenciesSet;
  ScAddrVector componentCurrentDependencies;
  while (componentDependenciesIterator->Next())
  {
    componentDependenciesSet = componentDependenciesIterator->Get(2);
    componentCurrentDependencies =
        utils::IteratorUtils::getAllWithType(context, componentDependenciesSet, ScType::NodeConst);
    componentDependencies.insert(
        componentDependencies.cend(), componentCurrentDependencies.cbegin(), componentCurrentDependencies.cend());
  }
```